### PR TITLE
Fixes issue 184, response headers to cofigure browsers' behaviour

### LINKED ISSUE #184: Get us from F to a better grade: https://observatory.mozilla.org/analyze/start.microprofile.io
https://observatory.mozilla.org/analyze/start.microprofile.io

### PR DESCRIPTION
**Pull request by Karm**
Hello,
fixes #184 
# Web page behaviour

This PR gets us from grade F to grade B without braking Primefaces. I didn't figure out how to get rid of inline JavaScript without breaking PrimeFaces, so we are probably stuck with B :smile: 

See https://observatory.mozilla.org/analyze/starter2.karms.biz

### CODE REVIEW

**Comment by rdebusscher** [comment]
Thx @Karm 

For the display of the curl command, create a separate issue so that we do not mix several issues in 1 PR.

I'll have a look at the PR later and propose a better alternative for the many
`<f:event type="preRenderView"`

File: src/main/webapp/index.xhtml [hunk 1]
```diff
@@ -34,6 +34,19 @@
       xmlns:pt="http://xmlns.jcp.org/jsf/passthrough"
 >
 
+<f:event type="preRenderView"
```
Comment (rdebusscher) [inline src/main/webapp/index.xhtml]: For setting the response headers, this is more readable and more performant:

```
@WebFilter(urlPatterns = "*.xhtml")
public class HeaderFilter implements Filter {

    @Override
    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
        HttpServletResponse servletResponse = (HttpServletResponse) response;
        servletResponse.setHeader("X-XSS-Protection", "1; mode=block");
        ....
        chain.doFilter(request, response);
    }
...
}
```

File: src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -37,6 +37,19 @@
...
+    
+    <context-param>
+        <param-name>primefaces.CSP</param-name>
+        <param-value>true</param-value>
+    </context-param>
+
+    <session-config>
+        <cookie-config>
+            <http-only>true</http-only>
+            <!--Breaks PrimeFaces: <secure>true</secure>-->
```
Comment (rdebusscher) [inline src/main/webapp/WEB-INF/web.xml]: or because SSL termination is not done by the server.

File: src/main/webapp/index.xhtml (hunk 1 above)
Comment (Karm) [reply src/main/webapp/index.xhtml]: @rdebusscher Thank! Will do.

File: src/main/webapp/WEB-INF/web.xml (hunk 2 above)
Comment (Karm) [reply src/main/webapp/WEB-INF/web.xml]: @rdebusscher True. I have Nginx there on my box doing a proxy.

**Comment by Karm** [comment]
@rdebusscher Ready for review & merge.

**Comment by Karm** [comment]
Had acks. Merged.

**Pull request merged**